api/src/main/java/org/openmrs/api/db/hibernate/HibernateAdministrationDAO.java

```diff
@@ -200,9 +202,51 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
 	
 	/**
 	 * @see org.openmrs.api.db.AdministrationDAO#validate(java.lang.Object, Errors)
+	 * @should pass validation if all required fields have proper values
 	 */
 	@Override
 	public void validate(Object object, Errors errors) throws DAOException {
+		String name = object.getClass().getSimpleName();
+		Class entityClass = object.getClass();
+		for (Field field : object.getClass().getDeclaredFields()) {
```

Comment:
Why don't you use hibernate's ClassMetadata, otherwise getDeclaredFields wouldn't return inherited fields and also not all fields defined on a class are persistent.